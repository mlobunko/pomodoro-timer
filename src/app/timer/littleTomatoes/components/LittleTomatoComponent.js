import React from "react";
import PropTypes from "prop-types";

import { StyledAnimatedLittleTomato, StyledLittleTomato } from "./styles";

export const LittleTomatoComponent = ({
  isWorkingTime,
  numberOfLittleTomatoes,
  order,
  timerOn
}) => (
  <React.Fragment>
    {timerOn && isWorkingTime && numberOfLittleTomatoes === order ? (
      <StyledAnimatedLittleTomato />
    ) : (
      numberOfLittleTomatoes > order && <StyledLittleTomato />
    )}
  </React.Fragment>
);

LittleTomatoComponent.displayName = "LittleTomato";

export default LittleTomatoComponent;

LittleTomatoComponent.propTypes = {
  isWorkingTime: PropTypes.bool.isRequired,
  numberOfLittleTomatoes: PropTypes.number.isRequired,
  order: PropTypes.number.isRequired,
  timerOn: PropTypes.bool.isRequired
};
