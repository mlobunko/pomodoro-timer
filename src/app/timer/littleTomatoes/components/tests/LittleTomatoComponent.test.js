import React from "react";
import { shallow } from "enzyme";

import { LittleTomatoComponent } from "../LittleTomatoComponent";

const props = {
  isWorkingTime: true,
  numberOfLittleTomatoes: 2,
  order: 2,
  timerOn: true
};

describe("render", () => {
  it("StyledAnimatedLittleTomato", () => {
    const wrapper = shallow(<LittleTomatoComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("StyledLittleTomato", () => {
    const wrapper = shallow(
      <LittleTomatoComponent {...props} numberOfLittleTomatoes={3} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("empty, timerOn false", () => {
    const wrapper = shallow(
      <LittleTomatoComponent {...props} timerOn={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("empty, isWorkingTime false", () => {
    const wrapper = shallow(
      <LittleTomatoComponent {...props} isWorkingTime={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
