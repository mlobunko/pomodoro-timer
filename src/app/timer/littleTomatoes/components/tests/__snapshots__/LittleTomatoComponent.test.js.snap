// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render StyledAnimatedLittleTomato 1`] = `
<Fragment>
  <StyledAnimatedLittleTomato />
</Fragment>
`;

exports[`render StyledLittleTomato 1`] = `
<Fragment>
  <StyledLittleTomato />
</Fragment>
`;

exports[`render empty, isWorkingTime false 1`] = `<Fragment />`;

exports[`render empty, timerOn false 1`] = `<Fragment />`;
